### PR TITLE
[ci] Add build job for Ubuntu ARM

### PR DESCRIPTION
**Pull request by gjulianm**
### What and why?

This PR adds a Github job that creates standalone binaries for ARM64 

### How?

Using the Github runners in ARM, I duplicated the `build-binary-ubuntu` job to generate ARM64 binaries.

### Review checklist

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -147,6 +147,30 @@ jobs:
       - name: Test generated standalone binary
         run: yarn dist-standalone:test
 
+  standalone-binary-test-ubuntu-arm:
+    name: Test standalone binary in ARM ubuntu
+    runs-on:
+      group: ARM LINUX SHARED
+      labels: arm-4core-linux
```
Comment (Drarig29) [inline .github/workflows/ci.yml:154]: Apparently, [other Datadog repos](https://github.com/search?q=org%3ADataDog+%22arm-4core-linux%22&type=code) use the following syntax:
```suggestion
    runs-on: arm-4core-linux
```

Comment (gjulianm) [reply .github/workflows/ci.yml:154]: I'm using this syntax as it's the one the folks at #build-support recommended ([docs](https://datadoghq.atlassian.net/wiki/spaces/DEVX/pages/3790274750/Github+Actions)), in theory this should be more robust against changes

**Comment by gjulianm** [comment]
By the way, based on [this job result](https://github.com/DataDog/datadog-ci/actions/runs/9450981908/job/26030865507?pr=1333) it seems like the macOS runners are running ARM64 too, but the naming of the binary implies it's x64.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Drarig29) [reply .github/workflows/ci.yml:154]: Thanks, I wasn't aware of these docs 👍

**Comment by Drarig29** [comment]
@gjulianm oh, thanks. The naming used to be correct, but it became incorrect recently because `macos-latest` went from x64 to arm64 (see https://github.com/DataDog/datadog-ci/pull/1291 for a related fix).

To bring back a x64 macOS binary, we'd need another job - on `macos-latest-large` (see [docs](https://github.com/actions/runner-images)).

But since nobody complained, I guess the macOS binary isn't used, so let's just align the naming and keep arm64 only.

**Comment by gjulianm** [comment]
> @gjulianm oh, thanks. The naming used to be correct, but it became incorrect recently because `macos-latest` went from x64 to arm64 (see #1291 for a related fix).
> 
> To bring back a x64 macOS binary, we'd need another job - on `macos-latest-large` (see [docs](https://github.com/actions/runner-images)).
> 
> But since nobody complained, I guess the macOS binary isn't used, so let's just align the naming and keep arm64 only.

Do you want me to do it in this PR? It's a quick change so it's no problem

**Comment by Drarig29** [comment]
@gjulianm cool, then I'm not against it! Thanks 🙇

**Pull request merged**